orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -64,7 +66,9 @@ class CompleteStageHandler(
             // check to see if this stage has any unplanned synthetic after stages
             var afterStages = stage.firstAfterStages()
             if (afterStages.isEmpty()) {
-              stage.planAfterStages()
+              stage.withAuth {
```

Comment:
`StartStageHandler` was already doing this which is why `DeployCanaryStage` half-worked.